agent.go

```diff
@@ -53,33 +53,36 @@ type process struct {
 type container struct {
 	sync.RWMutex
 
-	id          string
-	initProcess *process
-	container   libcontainer.Container
-	config      configs.Config
-	processes   map[string]*process
-	mounts      []string
+	id              string
+	initProcess     *process
+	container       libcontainer.Container
+	config          configs.Config
+	processes       map[string]*process
+	mounts          []string
+	useSandboxPidNs bool
```

Comment:
Can you drop this `useSandboxPidNs` field? As stated in my earlier comment, I think it can be determined by the container spec instead:
 - no NEWPID: the container joins the sandbox level pidns
 - NEWPID with empty pidns path: the container lives in its own pidns
 - NEWPID with non-empty pidns path: the container joins the sandbox level pidns

Reply:
I added this field since currently we are clearing out the namespace paths in the runtime before they are passed to the agent. I suppose this was done so as to not try to use the host namespaces accidentally in the agent (which we may need to check for mount, user namespaces?).
I also like the fact that this configuration is sent explicitly by the runtime. The runtime actually checks that the namespace for a container is the same as the very first container, and then sets the `useSandboxPidNs` field. This way we check that it is not just some other namespace path on the host. (Have raised a separate PR for this in the runtime.)
Whats do you think? Do you see any issues in passing this field explicitly?

Reply:
Fair enough. I thought it duplicates with config.NEWPID but kata runtime is the only user and it clears the namespace path, I agree that explicit setting is better. Besides we are already diverging from OCI spec in the namespace handling, it is better not to use the spec to avoid confusion. Would you please add some comments on the proto change to make it clear that the field overrides whatever is set in config.NEWPID?

> which we may need to check for mount, user namespaces

Yes, we should. All namespace paths in the config should be cleared IMO. Otherwise they are potential attach surfaces.